### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by husush23**
### In project I have used and practiced the following technologies :

- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions, and reducers in React.

### Furthermore, I have implemented some functionalities for this project including:
- Fetched data from API
- filtered and the user can search individual pair-currencies
- User can click and see more details about the pairs
- Written test for all components

I would like to hear👂🏻 from you and would be proud to see project approved✅
[Here is presentation video](https://www.loom.com/share/b29829f6fa254d5894319f6239fec90b)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,205 @@
...
+- **Fetches API data**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+[Not yet !](url)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (mikethreels) [inline README.md]: - [ ] according to the requirements you should deploy the application using Render, Netlify or GitHub Pages and a link to the deployed project should be present in the README.
- [ ] you should have recorded a video presenting the features of the project, please take the below requirements into consideration when creating the video:
  - The maximum length of the video is 5 minutes.
  - The video should contain a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
  - The video should contain a demo of the project, highlighting all the project features.
  - The video should contain a highlight about a piece of the code you find interesting or are proud of making.

File: README.md [hunk 2]
```diff
@@ -1 +1,205 @@
...
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to express my heartfelt gratitude to all the contributors who have helped shape and improve this project. Your dedication, insights, and efforts have been invaluable in making this project what it is today. Thank you for your contributions and for being part of this collaborative journey.
+
+Please note that the list of contributors may be available in the project's version control system, such as Git, where you can view the commit history and see all the individuals who have made contributions over time. Encouraging users to check the project's repository or documentation for a comprehensive list of contributors is a good practice.
+
```
Comment (mikethreels) [inline README.md]: - [ ] according to the requirements the README should contain an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: README.md [hunk 3]
```diff
@@ -1 +1,205 @@
...
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (mikethreels) [inline README.md:203]: - [ ] The link to the license is not working as expected please make sure it is working as presented in the Microverse template.

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchCrypto());
+  }, [dispatch]);
+  return (
+    <div className="App">
+      <BrowserRouter>
+        <Routes>
+          <Route path="/" element={<Currencies />} />
```
Comment (mikethreels) [inline src/App.js:18]: - [x] According to the requirements the Layout should be the same as the example which includes composition and space between elements. Have a look at the below examples and please make sure to make your project more similar to the original:
original:
afbeelding
your project:
afbeelding

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Hussein, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job making the project work.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 4 above)
Comment (husush23) [reply src/App.js:18]: I removed the margin between the  items and this the look, it looks the same the difference is the color and search box.
Screenshot 2023-07-06 095935

Comment (VICTOR-CHUKWU) [reply src/App.js:18]: - [ ] You did a good job implementing the changes required but the background of the card is not looking like the design on mobile, know that the card's background colour is supposed to be interchanged between cards. you can achieve this by giving all odd-numbered cards a light background and all even-numbered cards a darker background
in your design :point_down: 
Screenshot from 2023-07-06 09-59-24
given design :point_down: 
Screenshot from 2023-04-28 11-04-25

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @husush23 , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/App.js (hunk 4 above)
Comment (husush23) [reply src/App.js:18]: @VICTOR-CHUKWU You see that I gave the color every odd card a different color and also every even too, what is the other things you are saying please be clear

Comment (VICTOR-CHUKWU) [reply src/App.js:18]: so what is happening in mobile is that all odd cards are under each other so you can use this selector `*:nth-child(4n + 1)` to make the background diff

Comment (VICTOR-CHUKWU) [reply src/App.js:18]: i hope that works

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @husush23, 

To highlight 👍🏽 
✔️ Card colours are alternating
✔️ App is deployed
✔️ Video presentation is made

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**